hack/images/dcos/bootstrap.sh

```diff
@@ -220,70 +218,88 @@ exec 2>&1
 
 export KUBERNETES_MASTER="${kube_master}"
 
-/opt/kubectl get rc kube-dns-v4 >/dev/null && \
-  /opt/kubectl get service kube-dns >/dev/null && \
+/opt/kubectl get rc --namespace=kube-system -l k8s-app=kube-dns | grep kube-dns >/dev/null && \
+  /opt/kubectl get service --namespace=kube-system kube-dns >/dev/null && \
   touch kill && exit 0
 
-for i in $obj; do
-  /opt/kubectl create -f ${sandbox}/\$i
-done
+/opt/kubectl create -f ${sandbox}/skydns-rc.yaml
+/opt/kubectl create -f ${sandbox}/skydns-svc.yaml
 EOF
 
   sed -i -e '$i test -f kill && exec s6-svc -d $(pwd) || exec \\' ${service_dir}/kube_dns/finish
 
-  prepare_service_depends apiserver ${kube_master}/healthz ok kube_dns
+  apiserver_depends="${apiserver_depends} kube_dns"
 }
 
+# launch kube-dns if enabled
 kube_cluster_dns=""
 kube_cluster_domain=""
-# launch kube-dns if enabled
-if test -n "$ENABLE_DNS"; then
+if [ "${ENABLE_DNS:-true}" == true ]; then
```

Comment:
What do you mean? I kept the style to pass the default in place. One could that once for all variables at the top. But that's another refactoring.